builtin.py

```diff
@@ -11,13 +12,17 @@ def summary(self):
 # Base node classes
 
 class Node:
-    def __init__(self, title, desc):
+    def __init__(self, title: str, desc: str, connections: list[Node] = []):
```

Comment:
Great addition! I agree, type checking in Python is definitely something we should enforce moving forward.